pkg/api/v1alpha1/cluster_types.go

```diff
@@ -832,11 +832,48 @@ var validCiliumPolicyEnforcementModes = map[CiliumPolicyEnforcementMode]bool{
 	CiliumPolicyModeNever:   true,
 }
 
+// Reasons for the terminal failures while reconciling the Cluster object.
+const (
+	// MissingDependentObjectsReason reports that the Cluster is missing dependent objects.
+	MissingDependentObjectsReason = "MissingDependentObjects"
+
+	// ManagementClusterRefInvalidReason reports that the Cluster management cluster reference is invalid. This
+	// can whether if it does not exist or the cluster referenced is not a management cluster.
+	ManagementClusterRefInvalidReason = "ManagementClusterRefInvalid"
+
+	// ClusterSpecValidationFailedReason reports that the Cluster spec validation has failed.
+	ClusterSpecValidationFailedReason = "ClusterSpecValidationFailed"
```

Comment:
instead of validation failed, can we just say `...Invalid`? same comment for the other reasons

Also I don't think we need to specify `ClusterSpec` right? Can it just be `ClusterInvalid`?